app/assets/javascripts/angular/controllers/dashboard_ctrl.js

```diff
@@ -74,8 +74,18 @@ angular.module("Prometheus.controllers")
     $scope.nextCycleRedraw();
   };
 
+  $window.history.replaceState({initial: true}, "Promdash - " + dashboardName, $window.location.href);
   $document.keydown(function(ev) {
-    if (ev.keyCode === 27) { // Escape keycode
+    var Z_KEY = 90;
+    var ESC_KEY = 27;
+    if (ev.keyCode === Z_KEY && document.activeElement.tagName !== "INPUT") {
```

Comment:
Is it necessary to check for input? Will a keypress really trigger the event otherwise? I'd have expected that the browser doesn't forward keypress events anyway in that situation.